lib/process.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs');
+
+module.exports = {
+  uploadFieldFiles: async function (req, name, files, {
+    warn,
+    insert
+  }) {
+
+    // Upload each file for the field, then join IDs.
+    try {
+      // Field names can't include
+      const ids = [];
+
+      for (const entry in files) {
+        if (!matchesName(entry, name)) {
+          continue;
+        }
+
+        const attachment = await insert(req, files[entry], {
+          permissions: false
+        });
+
+        ids.push(attachment._id);
+      }
+
+      return ids;
+    } finally {
+      for (const file of (Object.values(req.files || {}))) {
+        if (!matchesName(file.fieldName, name)) {
```

Comment:
I think this logic belongs in the route, so it fires no matter whether this method is ever called or not. Also it shouldn't care if the field was relevant or not... any temp files lying around from the middleware must die!